ui/app/controllers/vault/cluster/settings/mount-secret-backend.js

```diff
@@ -1,30 +1,34 @@
 import { inject as service } from '@ember/service';
 import Controller from '@ember/controller';
 import { supportedSecretBackends } from 'vault/helpers/supported-secret-backends';
+import { allEngines } from 'vault/helpers/mountable-secret-engines';
+import { action } from '@ember/object';
 
 const SUPPORTED_BACKENDS = supportedSecretBackends();
 
-export default Controller.extend({
-  wizard: service(),
-  actions: {
-    onMountSuccess: function (type, path) {
-      let transition;
-      if (SUPPORTED_BACKENDS.includes(type)) {
-        if (type === 'kmip') {
-          transition = this.transitionToRoute('vault.cluster.secrets.backend.kmip.scopes', path);
-        } else if (type === 'keymgmt') {
-          transition = this.transitionToRoute('vault.cluster.secrets.backend.index', path, {
-            queryParams: { tab: 'provider' },
-          });
-        } else {
-          transition = this.transitionToRoute('vault.cluster.secrets.backend.index', path);
-        }
+export default class MountSecretBackendController extends Controller {
+  @service wizard;
+  @service router;
+
+  @action
+  onMountSuccess(type, path) {
+    let transition;
+    if (SUPPORTED_BACKENDS.includes(type)) {
+      const { engineRoute } = allEngines().findBy('type', type);
+      if (engineRoute) {
+        transition = this.router.transitionTo(`vault.cluster.secrets.backend.${engineRoute}`, path);
+      } else if (type === 'keymgmt') {
+        transition = this.router.transitionTo('vault.cluster.secrets.backend.index', path, {
+          queryParams: { tab: 'provider' },
```

Comment:
I like how you moved the `engineRoute` to the helper level, should we do the same with `queryParams`? I imagine more of our future engines we'll want to land on the `overview` page

Reply:
Yes great suggestion! This will clean up the route further and also make it more flexible for future engines that still use the query param pattern for the tabs.